openedx/core/lib/js_utils.py

```diff
@@ -68,3 +52,32 @@ def escape_json_dumps(obj, cls=EdxJSONEncoder):
     encoded_json = json.dumps(obj, ensure_ascii=True, cls=cls)
     encoded_json = _escape_json_for_html(encoded_json)
     return encoded_json
+
+
+def escape_js_str(js_str):
+    """
+    Escape a javascript string that is safe to be embedded in HTML.
+
+    Usage:
+        Can be used inside a Mako template inside a <SCRIPT> as follows:
+            var my_js_str = "${escape_js_str(my_js_str) | n}"
+
+        Must include the surrounding quotes for the string.
+
+        Use the "n" Mako filter above.  It is possible that the
+            default filter may include html encoding in the future, and
```

Comment:
I'll adjust comments to follow Ned's Confluence comment around using "escaping" for this type of protection, and "encoding" for something like unicode.